jolly-jellyfish/src/page_maker/tests.py

```diff
@@ -38,13 +50,115 @@ def test_register_and_home(self):
         self.assertNotIn(USERNAME.encode(), res, 'Our username was already in the home view.')
 
 
+class UserViewsTest(TestCase):
+    fixtures = ['users.yaml', 'webpages.yaml', 'templates.yaml']
+
+    def test_user_detail(self):
+        user = User.objects.get(pk=1)
+        res = self.client.get('/users/1/')
+        self.assertEqual(res.status_code, 200)
+        self.assertEqual(res.context['viewed_user'], user)
+        self.assertIsInstance(res.context['user_pages'][0], Webpage)
+
+    def test_user_update(self):
+        self.client.login(username=USERNAME, password=PASSWORD)
+        res = self.client.post('/users/2/update', {
+            'first_name': 'john', 'last_name': 'lemon', 'email': 'jl@sth.com'
+        })
+        self.assertRedirects(res, '/')
+        user = User.objects.get(pk=2)
+        self.assertEqual(user.last_name, 'lemon')
+
+    def test_user_delete(self):
+        self.client.login(username=USERNAME, password=PASSWORD)
+        res = self.client.post('/users/2/delete')
+        self.assertRedirects(res, '/')
+        self.assertEqual(User.objects.count(), 1)
+
+
+class WebpageViewsTest(TestCase):
+    fixtures = ['users.yaml', 'templates.yaml', 'webpages.yaml', 'comments.yaml']
+
+    def test_webpage_create_redirect(self):
+        res = self.client.get('/pages/new')
+        self.assertRedirects(res, '/users/login?next=/pages/new')
+
+    def test_webpage_create(self):
+        self.client.login(username=USERNAME, password=PASSWORD)
+        res = self.client.get('/pages/new')
+        self.assertEqual(res.status_code, 200)
+        self.assertTemplateUsed(res, template_name='page_maker/webpage_create.html')
```

Comment:
Are we sure we want to hardcode the template used? This might make for a flakey test (as in: a test we will need to change eventually if we rename templates or something).

Though given our dependence on certain items being in the `res.context` this might not be so important...